packages/financial-templates-lib/src/price-feed/BalancerPriceFeed.js

```diff
@@ -1,13 +1,30 @@
 const assert = require("assert");
 const { PriceFeedInterface } = require("./PriceFeedInterface");
 const { BlockHistory, PriceHistory } = require("./utils");
+const { parseFixed } = require("@ethersproject/bignumber");
+const { MAX_SAFE_JS_INT } = require("@uma/common");
 
-// Gets balancer prices as well as historical prices
+// Gets balancer spot and historical prices. This price feed assumes that it is returning
+// prices as 18 decimals of precision, so it will scale up the pool's price as reported by Balancer contracts
+// if the user specifies that the Balancer contract is returning non-18 decimal precision prices.
 class BalancerPriceFeed extends PriceFeedInterface {
-  constructor(logger, web3, getTime, abi, address, tokenIn, tokenOut, lookback) {
+  constructor(
+    logger,
+    web3,
+    getTime,
+    abi,
+    address,
+    tokenIn,
+    tokenOut,
+    lookback,
+    twapLength,
+    poolDecimals = 18,
```

Comment:
This change (and the `decimals`) field below will need to change slightly following https://github.com/UMAprotocol/protocol/pull/2276/files getting merged. Specifically, the price feed decimals should return a price scaled in the same way that the DVM would respond to a price request. It's not a large change but does have a small impact on the interface and how you define the scaling within the feed.

Reply:
Cool makes sense I'll hold off on merging until #2276 is merged